src/platform/ModulesManager.c

```diff
@@ -0,0 +1,742 @@
+    }
+    else if (NULL == (config = json_parse_file(configJson)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse configuration JSON (%s)", configJson);
+    }
+    else if (NULL == (configObject = json_value_get_object(config)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to get config object");
+    }
+    else if (0 == (version = json_object_get_number(configObject, g_modelVersion)))
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to get model version from configuration JSON (%s)", configJson);
+    }
+    else
+    {
+        // "Azure OSConfig <version>;<osconfig version>" + null-terminator
+        clientNameSize = strlen(AZURE_OSCONFIG) + strlen(OSCONFIG_VERSION) + 5;
+
+        if (NULL == (clientName = (char*)malloc(clientNameSize)))
+        {
+            OsConfigLogError(GetPlatformLog(), "Failed to allocate memory for client name");
+        }
+        else
+        {
+            memset(clientName, 0, clientNameSize);
+            snprintf(clientName, clientNameSize, "%s %d;%s", AZURE_OSCONFIG, version, OSCONFIG_VERSION);
+        }
+
+        OsConfigLogInfo(GetPlatformLog(), "Loading modules from '%s' (%s)", directory, clientName);
+
+        while (NULL != (entry = readdir(dir)))
+        {
+            if (NULL == entry->d_name)
+            {
+                continue;
+            }
+
+            if ((strcmp(entry->d_name, "") == 0) || (strcmp(entry->d_name, ".") == 0) || (strcmp(entry->d_name, "..") == 0))
+            {
+                continue;
+            }
+
+            if (NULL == strstr(entry->d_name, MODULE_EXT))
+            {
+                continue;
+            }
+
+            pathSize = strlen(directory) + strlen(entry->d_name) + 2;
+
+            if (NULL == (path = malloc(pathSize)))
+            {
+                continue;
+            }
+
+            memset(path, 0, pathSize);
+            snprintf(path, pathSize, "%s/%s", directory, entry->d_name);
+
+            if (NULL != (module = LoadModule(clientName, path)))
+            {
+                module->next = g_modules;
+                g_modules = module;
+            }
+            else
+            {
+                OsConfigLogError(GetPlatformLog(), "Failed to load module: %s", entry->d_name);
+            }
+
+            FREE_MEMORY(path);
+        }
+    }
+
+    if (config && configObject)
+    {
+        if (NULL != (reportedArray = json_object_get_array(configObject, g_reportedObjects)))
+        {
+            reportedCount = (int)json_array_get_count(reportedArray);
+
+            if (0 < reportedCount)
+            {
+                reportedSize = sizeof(REPORTED_OBJECT) * reportedCount;
+
+                if (NULL != (reported = (REPORTED_OBJECT*)malloc(reportedSize)))
+                {
+                    memset(reported, 0, reportedSize);
+
+                    for (; i < reportedCount; i++)
```

Comment:
>for (; i < reportedCount; i++) [](http://example.com/codeflow?start=20&length=30)

No no no.... Still add the i init here, just do not declare it here. Like this:

for (i = 0; i < ....